https://github.com/linuxmint/cinnamon/commit/9ccce54c29f7c78404e8819d7be7a051efff4df1

From 9ccce54c29f7c78404e8819d7be7a051efff4df1 Mon Sep 17 00:00:00 2001
From: Eli Schwartz <eschwartz@archlinux.org>
Date: Thu, 3 Feb 2022 09:09:24 -0500
Subject: [PATCH] gtkdoc: remove dependencies on custom target files (#10606)

Sadly, the `dependencies` kwarg does not actually do what it seems to be
trying to be used for, here. It is for listing dependency or library
objects whose compiler flags should be added to gtkdoc-scangobj.

It will not actually add ninja target dependencies. The similar kwarg in
other meson functions (e.g. genmarshal and compile_schemas) that *do*
allow adding target dependencies, is `depend_files`.

Older versions of meson simply did nothing in an if/elif/elif block
where these custom_targets never matched anything, and were thus
silently ignored.

Meson 0.61 type-validates the arguments and rejects CustomTarget as
invalid:

```
docs/reference/cinnamon-js/meson.build:11:6: ERROR: gnome.gtkdoc keyword argument 'dependencies' was of type array[CustomTarget] but should have been array[Dependency | SharedLibrary | StaticLibrary]
```
---
 docs/reference/cinnamon-js/meson.build | 1 -
 1 file changed, 1 deletion(-)

diff --git a/docs/reference/cinnamon-js/meson.build b/docs/reference/cinnamon-js/meson.build
index 7c96c875a0..66a5e156eb 100644
--- a/docs/reference/cinnamon-js/meson.build
+++ b/docs/reference/cinnamon-js/meson.build
@@ -12,7 +12,6 @@ gnome.gtkdoc(
     'cinnamon-js',
     mode: 'xml',
     main_xml: 'cinnamon-js-docs.sgml',
-    dependencies: parts_files,
     src_dir: meson.current_build_dir(),
     install: true,
 )
